sdk/src/Services/DynamoDBv2/Custom/DynamoDBEnumerations.cs

```diff
@@ -0,0 +1,36 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ * 
+ *  http://aws.amazon.com/apache2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+namespace Amazon.DynamoDBv2
+{
+    /// <summary>
+    /// The higher-level programming models rely on an internal cache of the DynamoDB table's metadata 
+    /// to construct and validate requests. This controls how the cache key is derived, which influences 
+    /// when the SDK will call <see cref="IAmazonDynamoDB.DescribeTable(string)"/> internally to populate the cache.
+    /// </summary>
+    public enum MetadataCachingMode
+    {
+        /// <summary>
+        /// The cache key will be a combination of the table name, credentials, region and service URL.
```

Comment:
Add something like
```
This ensures applications using tables with the same name but different definitions have their own metadata cached in the application. This will require additional `DescribeTable` API calls as credentials are refreshed.
```

Reply:
Added in https://github.com/aws/aws-sdk-net/compare/1161dc30f02a5a18108c28b0191f21da5080a368..0c2afdec6c32efc6ab9ebd3841a433bf19683181